src/util/audio/VorbisProducer.cpp

```diff
@@ -32,21 +33,32 @@ bool VorbisProducer::start(std::string filename, unsigned int timestamp)
 	this->audioQueue->setAudioAttributes(this->sfInfo.samplerate, static_cast<unsigned int>(this->sfInfo.channels));
 
 	this->producerThread = new std::thread(
-			[&, filename]
+			[&, filename, seekPosition]
 			{
 				size_t numFrames = 1;
 				auto sampleBuffer = new float[1024 * this->sfInfo.channels];
 				std::unique_lock<std::mutex> lock(audioQueue->syncMutex());
-
+				
+				size_t position = seekPosition;
+				
 				while (!this->stopProducer && numFrames > 0 && !this->audioQueue->hasStreamEnded())
 				{
-			        numFrames = sf_readf_float(this->sfFile, sampleBuffer, 1024);
+					numFrames = sf_readf_float(this->sfFile, sampleBuffer, 1024);
+					position+=numFrames;
 
 					while (this->audioQueue->size() >= this->sample_buffer_size && !this->audioQueue->hasStreamEnded() && !this->stopProducer)
 					{
 						audioQueue->waitForConsumer(lock);
 					}
 
+					if (this->seekSeconds != 0)
+					{
+						position += seekSeconds * this->sfInfo.samplerate;
+						if (position < 0) position = 0;
+						sf_seek(this->sfFile, position, SEEK_SET);
```

Comment:
Using `SEEK_CUR` you can just apply the offset which is then used from the current playback position. That saves you these awkward position calculations all throughout the code.

Reply:
Damn it really was that simple? I'm gonna try this out when I get home.

Reply:
Yeah :joy: 
http://www.mega-nerd.com/libsndfile/api.html#seek